x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -630,6 +668,11 @@ ClientDataFrameIndexerBuilder setNextCheckpoint(DataFrameTransformCheckpoint nex
 
         @Override
         protected void onStart(long now, ActionListener<Boolean> listener) {
+            if (transformTask.taskState.get() == DataFrameTransformTaskState.FAILED) {
```

Comment:
All of these failure state checks on the method callbacks are to cover for the situation where another trigger sneaks in after we fail the first one. We want that second trigger to fail ASAP to prevent undesired state interactions.